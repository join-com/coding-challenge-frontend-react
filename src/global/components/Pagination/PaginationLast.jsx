import React from 'react';
import PropTypes from 'prop-types';

import PaginationTab from './PaginationTab';

import './PaginationLast.css';

const PaginationLast = ({
  totalPages, pageNumber, onPageNumberChange,
}) => {
  if (totalPages > 1 && pageNumber < totalPages) {
    return (
      <div className="PaginationLast">
        <div className="PaginationLast__container">
          <PaginationTab
            className="PaginationLastTab"
            currentPageNumber={totalPages}
            pageNumber={pageNumber}
            onPageNumberChange={onPageNumberChange}
          />
        </div>
      </div>
    );
  }
  return null;
};

export default PaginationLast;

PaginationLast.propTypes = {
  totalPages: PropTypes.number,
  pageNumber: PropTypes.number,
  onPageNumberChange: PropTypes.func,
};

PaginationLast.defaultProps = {
  totalPages: 0,
  pageNumber: 1,
  onPageNumberChange: () => {},
};
