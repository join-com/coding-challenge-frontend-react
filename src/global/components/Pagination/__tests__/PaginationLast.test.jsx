import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PaginationLast from '../PaginationLast';

const props = {
  totalPages: 3,
  pageNumber: 2,
};

describe('PaginationLast', () => {
  it('should render the component for totalPages > 1 & pageNumber < totalPages', () => {
    const tree = shallow(<PaginationLast {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component for pageNumber = totalPages', () => {
    const tree = shallow(<PaginationLast {...props} pageNumber={3} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component for totalPages = 1', () => {
    const tree = shallow(<PaginationLast {...props} totalPages={1} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
