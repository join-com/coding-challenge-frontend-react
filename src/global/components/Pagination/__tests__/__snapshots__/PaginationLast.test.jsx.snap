// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationLast should not render the component for pageNumber = totalPages 1`] = `""`;

exports[`PaginationLast should not render the component for totalPages = 1 1`] = `""`;

exports[`PaginationLast should render the component for totalPages > 1 & pageNumber < totalPages 1`] = `
<div
  className="PaginationLast"
>
  <div
    className="PaginationLast__container"
  >
    <PaginationTab
      className="PaginationLastTab"
      currentPageNumber={3}
      onPageNumberChange={[Function]}
      pageNumber={2}
    />
  </div>
</div>
`;
